lib/rubocop/cop/lint/ambiguous_regexp_literal.rb

```diff
@@ -21,21 +21,20 @@ module Lint
       #
       #   # With parentheses, there's no ambiguity.
       #   do_something(/pattern/i)
-      class AmbiguousRegexpLiteral < Cop
+      class AmbiguousRegexpLiteral < Base
         include ParserDiagnostic
+        extend AutoCorrector
 
         MSG = 'Ambiguous regexp literal. Parenthesize the method arguments ' \
               "if it's surely a regexp literal, or add a whitespace to the " \
               'right of the `/` if it should be a division.'
 
-        def autocorrect(node)
-          lambda do |corrector|
-            add_parentheses(node, corrector)
-          end
-        end
-
         private
 
+        def autocorrect(corrector, node)
```

Comment:
I don't understand how this is supposed to work tbh... Who is calling `autocorrect`?